lib/create.js

```diff
@@ -15,22 +20,26 @@ const templateDir = './functionTemplates/nodejs/';
  * @param {string} functionPath - the path to create the function at
  * @returns {string} - the final name selected for the function
  */
-const create = async function create(functionName, functionPath) {
+const create = async function create(functionName, functionPath, runtime) {
   // removing the '-' from monikers string is required because they
   // don't allow the glue string to be empty.
   const finalName = functionName;
   validateName(finalName);
+  const template = runtimeToTemplateDir[runtime];
+  if (!template) {
+    throw new Error(`No template for runtime: ${runtime}`);
+  }
   // parse the templated yml and make the necessary modifications
-  const templatePath = path.join(__dirname, templateDir);
+  const templatePath = path.join(__dirname, templateDir, template);
   const binarisConf = await YMLUtil.loadBinarisConf(templatePath);
   const templateName = YMLUtil.getFuncName(binarisConf);
-  const funcConf = YMLUtil.getFuncConf(binarisConf, templateName);
+  const funcConf = { ...YMLUtil.getFuncConf(binarisConf, templateName), runtime };
   // replace the generic function name with the actual name
   YMLUtil.delFuncConf(binarisConf, templateName);
   YMLUtil.addFuncConf(binarisConf, finalName, funcConf);
   // now write out all the files that have been modified
   const file = funcConf.file;
-  await fse.copy(path.join(__dirname, templateDir, file), path.join(functionPath, file));
+  await fse.copy(path.join(__dirname, templateDir, template, file), path.join(functionPath, file));
```

Comment:
just do `join(templatePath, file)`

Reply:
I don't understand how this will work

Reply:
`const templatePath = path.join(__dirname, templateDir, template);`
`await fse.copy(path.join(__dirname, templateDir, template, file)....`

Fixed
`await fse.copy(path.join(templatePath, file)...`